server/src/main/java/com/linecorp/centraldogma/server/ZooKeeperServerConfig.java

```diff
@@ -41,12 +46,16 @@
      * @param quorumPort the quorum port number
      * @param electionPort the election port number
      * @param clientPort the client port number (0-65535)
+     * @param groupId the group ID to use hierarchical quorums
+     * @param weight the weight of the Zookeeper server
      */
     @JsonCreator
-    public ZooKeeperAddress(@JsonProperty(value = "host", required = true) String host,
-                            @JsonProperty(value = "quorumPort", required = true) int quorumPort,
-                            @JsonProperty(value = "electionPort", required = true) int electionPort,
-                            @JsonProperty(value = "clientPort", defaultValue = "0") int clientPort) {
+    public ZooKeeperServerConfig(@JsonProperty(value = "host", required = true) String host,
+                                 @JsonProperty(value = "quorumPort", required = true) int quorumPort,
+                                 @JsonProperty(value = "electionPort", required = true) int electionPort,
+                                 @JsonProperty(value = "clientPort", defaultValue = "0") int clientPort,
+                                 @JsonProperty("groupId") @Nullable Integer groupId,
+                                 @JsonProperty(value = "weight", defaultValue = "1") Integer weight) {
```

Comment:
Could we use `int` here given that we specified `defaultValue`?

Reply:
Initially, I tried to use `int`. However, the `defaultValue` did not work.
From the [Javadoc](https://fasterxml.github.io/jackson-annotations/javadoc/2.8/com/fasterxml/jackson/annotation/JsonProperty.html#defaultValue()), it is only used for documentation. 🤪

Reply:
Ah, that sounds unintuitive 😓

Reply:
Then should we also update other primitive parameters with `defaultValue`, e.g. `clientPort`?

Reply:
The default value of `clientPort` is 0. I thought we don't need to update it because the initialized value is 0. Anyway, let me check our codebase and update where `defaultValue` is used. 😀

Reply:
Fixed :-)